src/encoding/text.rs

```diff
@@ -1114,6 +1115,68 @@ mod tests {
         parse_with_python_client(encoded);
     }
 
+    #[test]
+    fn label_sets_can_be_composed() {
+        #[derive(Clone, Debug, Eq, Hash, PartialEq)]
+        struct Color(&'static str);
+        impl EncodeLabelSet for Color {
+            fn encode(
+                &self,
+                encoder: &mut crate::encoding::LabelSetEncoder,
+            ) -> Result<(), std::fmt::Error> {
+                use crate::encoding::EncodeLabel;
+                let Self(color) = *self;
+                let labels = ("color", color);
+                let encoder = encoder.encode_label();
+                labels.encode(encoder)
+            }
+        }
+
+        #[derive(Clone, Debug, Eq, Hash, PartialEq)]
+        struct Size(&'static str);
+        impl EncodeLabelSet for Size {
+            fn encode(
+                &self,
+                encoder: &mut crate::encoding::LabelSetEncoder,
+            ) -> Result<(), std::fmt::Error> {
+                use crate::encoding::EncodeLabel;
+                let Self(size) = *self;
+                let labels = ("size", size);
+                let encoder = encoder.encode_label();
+                labels.encode(encoder)
+            }
+        }
+
+        type Labels = (Color, Size);
+
+        let mut registry = Registry::default();
+        let family = Family::<Labels, Counter>::default();
+        registry.register("items", "Example metric", family.clone());
+
+        {
+            let labels = (Color("red"), Size("large"));
+            let counter = family.get_or_create(&labels);
+            counter.inc();
+        }
+        /*{
+            XXX: this test cannot deterministically check if multiple label values work. see #125.
+            let labels = (Color("blue"), Size("small"));
+            let counter = family.get_or_create(&labels);
+            counter.inc_by(2);
+        }*/
```

Comment:
Is any code-path of this new feature untested without this?

I guess we could always compare to both possible outcome strings. That said, I don't think it is necessary.

Reply:
okay, 98bca72 removes this.